clouddrift/kinematics.py

```diff
@@ -667,3 +724,147 @@ def velocity_from_position(
     dy /= dt
 
     return np.swapaxes(dx, time_axis, -1), np.swapaxes(dy, time_axis, -1)
+
+
+def spin(
+    u: np.ndarray,
+    v: np.ndarray,
+    time: np.ndarray,
+    difference_scheme: Optional[str] = "forward",
+    time_axis: Optional[int] = -1,
+) -> Union[float, np.ndarray]:
+    """Compute spin from velocities and times.
+
+    Spin is defined as (u'dv' - v'du') / (2 dt EKE) where u' and v' are
+    eddy-perturbations of the velocity field, EKE is eddy kinetic energy, dt is
+    the time step, and du' and dv' are velocity component increments during dt.
+    The result has a rank reduced by one relative to the input arrays.
+
+    u, v, and time can be multi-dimensional arrays. If the time axis, along
+    which the finite differencing is performed, is not the last one (i.e.
+    u.shape[-1]), use the time_axis optional argument to specify along which the
+    spin should be calculated. u, v, and time must either have the same shape,
+    or time must be a 1-d array with the same length as u.shape[time_axis].
+
+    Difference scheme can be one of three values:
+
+        1. "forward" (default): finite difference is evaluated as ``dx[i] = dx[i+1] - dx[i]``;
+        2. "backward": finite difference is evaluated as ``dx[i] = dx[i] - dx[i-1]``;
+        3. "centered": finite difference is evaluated as ``dx[i] = (dx[i+1] - dx[i-1]) / 2``.
+
+    Forward and backward schemes are effectively the same except that the
+    position at which the velocity is evaluated is shifted one element down in
+    the backward scheme relative to the forward scheme. In the case of a
+    forward or backward difference scheme, the last or first element of the
+    velocity, respectively, is extrapolated from its neighboring point. In the
+    case of a centered difference scheme, the start and end boundary points are
+    evaluated using the forward and backward difference scheme, respectively.
+
+    Reference for the definition of spin is:
```

Comment:
Put instead in a References section as in:

References
----------
.. [1] Veneziani, M., Griffa, A., Reynolds, A.M., Garraffo, Z.D. and Chassignet,
    E.P., 2005. Parameterizations of Lagrangian spin statistics and particle
    dispersion in the presence of coherent vortices. Journal of Marine Research,
    63(6), pp.1057-1083.
.. [2] if there is another one